### PR TITLE
Remove deprecated packages

### LINKED ISSUE #1308: Remove deprecated packages
As announced in https://github.com/deepmap/oapi-codegen/discussions/1309, we'll be removing the following deprecated packages:

  - `pkg/chi-middleware`
  - `pkg/fiber-middleware`
  - `pkg/gin-middleware`
  - `pkg/middleware`
  - `pkg/runtime`
  - `pkg/types`

### PR DESCRIPTION
**Pull request by jamietanna**
As part of #1308 and #1142, ahead of the v2 release, we want to remove
any deprecated packages to reduce the dependency tree, and nudge folks
to use the separated packages.

Closes #1308, #1142.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,13 +3,7 @@ module github.com/deepmap/oapi-codegen
...
 
 require (
-	github.com/apapsch/go-jsonmerge/v2 v2.0.0
 	github.com/getkin/kin-openapi v0.118.0
-	github.com/gin-gonic/gin v1.9.1
-	github.com/go-chi/chi/v5 v5.0.10
-	github.com/gofiber/fiber/v2 v2.49.1
-	github.com/google/uuid v1.3.1
-	github.com/kataras/iris/v12 v12.2.6-0.20230908161203-24ba4e8933b9
 	github.com/labstack/echo/v4 v4.11.1
```
Comment (jamietanna) [inline go.mod:7]: @deepmap-marcinr looks like this is pulled in through `pkg/testutil/request_helpers.go` - would you be happy us removing that package too? 

I know there are [a few of the other libraries](https://github.com/search?q=org%3Aoapi-codegen%20testutil&type=code) that depend on it, but I'm not sure how widespread it's used, and it'd be nice to remove this too, as then our direct dependency tree would be:

```
require (
	github.com/getkin/kin-openapi v0.118.0
	github.com/stretchr/testify v1.8.4
	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616
	golang.org/x/text v0.13.0
	golang.org/x/tools v0.12.0
	gopkg.in/yaml.v2 v2.4.0
)
```

Comment (deepmap-marcinr) [reply go.mod:7]: It's used everywhere in my company, however, we can move it to its own package, yes.

Comment (jamietanna) [reply go.mod:7]: Cool, thanks Marcin! I'll try and get the package set up in the next week or so 🤞🏽

Comment (jamietanna) [reply go.mod:7]: Raised https://github.com/deepmap/oapi-codegen/pull/1314 for the testutil migration

**Pull request merged**